packages/playwright-core/src/server/registry/index.ts

```diff
@@ -1188,9 +1188,11 @@ export class Registry {
       : `${displayName} playwright build v${descriptor.revision}`;
 
     const downloadFileName = `playwright-download-${descriptor.name}-${hostPlatform}-${descriptor.revision}.zip`;
-    const downloadConnectionTimeoutEnv = getFromENV('PLAYWRIGHT_DOWNLOAD_CONNECTION_TIMEOUT');
-    const downloadConnectionTimeout = +(downloadConnectionTimeoutEnv || '0') || 30_000;
-    await downloadBrowserWithProgressBar(title, descriptor.dir, executablePath, downloadURLs, downloadFileName, downloadConnectionTimeout).catch(e => {
+    // PLAYWRIGHT_DOWNLOAD_CONNECTION_TIMEOUT is a misnomer, it actually controls the socket's
+    // max idle timeout. Unfortunately, we cannot rename it without breaking existing user workflows.
+    const downloadSocketTimeoutEnv = getFromENV('PLAYWRIGHT_DOWNLOAD_CONNECTION_TIMEOUT');
+    const downloadSocketTimeout = +(downloadSocketTimeoutEnv || '0') || 30_000;
```

Comment:
```suggestion
    const downloadSocketTimeout = +(downloadSocketTimeoutEnv || '0') ?? 30_000;
```
lets fix this bug as a drive-by as well. Before passing `0` as an env var never worked.

Reply:
The fix you suggest will always set `downloadSocketTimeout` to `0` if the env variable is not provided (i.e. it would be equivalent to `const downloadSocketTimeout = +(downloadSocketTimeoutEnv || '0')`). I can address this issue separately from renaming (or not touch at all).